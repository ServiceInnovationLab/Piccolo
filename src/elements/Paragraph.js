import styled, { css } from 'styled-components';

const Paragraph = styled.p`
  font-size: 24px;
  color: #000;
  line-height: 1.5;
  ${props => props.small && css`
    font-size: 22px;
  `}
  ${props => props.medium && css`
    font-size: 24px;
  `}
  ${props => props.large && css`
    font-size: 34px;
  `}
  ${props => props.light && css`
    color: #fff;
  `}

  height: ${props => (props.adjust ? '100%' : 'auto')};

  @media(min-width: 1024px) {
    height: ${props => (props.adjust ? '55px' : 'auto')};
  }

  a {
    color: #2f3045;
  }
`;

export default Paragraph;
