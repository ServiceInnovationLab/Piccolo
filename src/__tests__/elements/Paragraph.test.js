import React from 'react';
import Paragraph from '../../elements/Paragraph';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

test('Paragraph default styles applied', () => {
  const tree = renderer.create(<Paragraph />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '24px');
  expect(tree).toHaveStyleRule('color', '#000');
  expect(tree).toHaveStyleRule('height', 'auto');
});

test('Paragraph small styles applied', () => {
  const tree = renderer.create(<Paragraph small />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '22px');
});

test('Paragraph medium styles applied', () => {
  const tree = renderer.create(<Paragraph medium />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '24px');
});

test('Paragraph large styles applied', () => {
  const tree = renderer.create(<Paragraph large />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('font-size', '34px');
});

test('Paragraph colour should be light if specified', () => {
  const tree = renderer.create(<Paragraph light />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
});

test('Paragraph height should be adjusted', () => {
  const tree = renderer.create(<Paragraph light adjust />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('color', '#fff');
  expect(tree).toHaveStyleRule('height', '55px');
});
