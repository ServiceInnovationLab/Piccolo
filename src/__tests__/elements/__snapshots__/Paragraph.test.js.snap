// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph colour should be light if specified 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  color: #fff;
  height: auto;
}

<p
  className="c0"
/>
`;

exports[`Paragraph default styles applied 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  height: auto;
}

<p
  className="c0"
/>
`;

exports[`Paragraph height should be adjusted 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  color: #fff;
  height: 55px;
}

<p
  className="c0"
/>
`;

exports[`Paragraph large styles applied 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  font-size: 34px;
  height: auto;
}

<p
  className="c0"
/>
`;

exports[`Paragraph medium styles applied 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  font-size: 24px;
  height: auto;
}

<p
  className="c0"
/>
`;

exports[`Paragraph small styles applied 1`] = `
.c0 {
  font-size: 24px;
  color: #000;
  font-size: 22px;
  height: auto;
}

<p
  className="c0"
/>
`;
